src/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.java

```diff
@@ -150,38 +164,41 @@ public void setPublicUpload(Boolean publicUpload) {
     @Override
     protected RemoteOperationResult run(OwnCloudClient client) {
         RemoteOperationResult result = null;
-        int status = -1;
+        int status;
 
         /// prepare array of parameters to update
-        List<Pair<String, String>> parametersToUpdate = new ArrayList<Pair<String, String>>();
+        List<Pair<String, String>> parametersToUpdate = new ArrayList<>();
+        if (mName != null) {
+            parametersToUpdate.add(new Pair<>(PARAM_NAME, mName));
+        }
         if (mPassword != null) {
-            parametersToUpdate.add(new Pair<String, String>(PARAM_PASSWORD, mPassword));
+            parametersToUpdate.add(new Pair<>(PARAM_PASSWORD, mPassword));
         }
         if (mExpirationDateInMillis < 0) {
             // clear expiration date
-            parametersToUpdate.add(new Pair(PARAM_EXPIRATION_DATE, ""));
+            parametersToUpdate.add(new Pair<>(PARAM_EXPIRATION_DATE, ""));
 
         } else if (mExpirationDateInMillis > 0) {
             // set expiration date
-            DateFormat dateFormat = new SimpleDateFormat(FORMAT_EXPIRATION_DATE);
+            DateFormat dateFormat = new SimpleDateFormat(FORMAT_EXPIRATION_DATE, Locale.GERMAN);
```

Comment:
```Locale.getDefault()``` would build something based in the language selection in the device. This date is not to be shown to the user, but to send to the server, so we must be sure of using a set of symbols appropriate. Could have been other country, just chose one I expected not to fail.